### DISCUSSION TITLE
how to  manually prevent the event from being passed to the client in the event hook handler?

### DISCUSSION THREAD

**Discussion by DreamMaoMao** [hyprwm/Hyprland#3632] [General]
![image](https://github.com/hyprwm/Hyprland/assets/30348075/71f9ebfd-061b-48e2-a4b7-ef753ed92377)

I am writing a plugin and I want to prevent the event from being passed to the client(if certain conditions are met). Does the current plugin system support this?

**Comment by vaxerski** [comment]
No, but you can function hook the `wlr_seat_pointer_notify_button` function and block it under conditions.

It would be a nice addition though, fair point.

**Reply by vaxerski** [reply]
done in https://github.com/hyprwm/Hyprland/commit/a61eb7694df25a75f45502ed64b1536fda370c1d

worth noting this breaks all plugins and they have to be fixed for this commit but there was no other way. Will probably not happen with the hookSystem ever again.

**Reply by DreamMaoMao** [reply]
thanks,i will try it later.

**Comment by DreamMaoMao** [comment]
![image](https://github.com/hyprwm/Hyprland/assets/30348075/0d5c1295-4fcb-4b8e-a244-41c4472c7e4a)
ok, it can work.
